invenio_drafts_resources/services/pid_manager.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""PIDManager."""
+
+from invenio_pidrelations.contrib.versioning import PIDNodeVersioning
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_pidstore.resolver import Resolver
+from invenio_records.api import Record
+
+from .minter import versioned_recid_minter_v2
+
+
+class PIDManagerConfig:
+    """PID manager config."""
+
+    record_cls = Record  # FIXME: This is duplicated with RecordServiceConfig
+    resolver_cls = Resolver
+    resolver_obj_type = "rec"
+    resolver_pid_type = "recid"  # PID type for resolver and fetch
+    minter_pid_type = "recid_v2"  # FIXME: Unused
+    draft_cls = None  # FIXME: This is duplicated with the user set one
+
+
+class PIDManager:
+    """PID manager."""
+
+    default_config = PIDManagerConfig
+
+    def __init__(self, config=None):
+        """Constructor."""
+        self.config = config or self.default_config
```

Comment:
This is definitely a pattern now :smile_cat: 

We will have to see how class configurations (versus instances) fare overtime.

Reply:
Yeah, I think we are sort of rendering useless `config.py`... on the other hand, users can still define classes, which I think is clearer than dicts (Biased developer opinion there hehe)

Reply:
I think so too. Skewing towards classes is better.